File: C#/tests/TurkeyHolidayValidatorTests.cs
using System;
using Xunit;

namespace HolidayChecker.Tests
{
    public class TurkeyHolidayValidatorTests
    {
        private TurkeyHolidayValidator validator;

        public TurkeyHolidayValidatorTests()
        {
            validator = new TurkeyHolidayValidator(false);
        }

        [Fact]
        public void TestTurkeyFixedHolidays()
        {
            Assert.True(validator.IsHoliday(new DateTime(2024, 4, 23)));
        }
    }
}

File: Swift/tests/TurkeyHolidayValidatorTests.swift
import XCTest
@testable import YourModuleName

class TurkeyHolidayValidatorTests: XCTestCase {
  var validator: TurkeyHolidayValidator!

  override func setUp() {
    super.setUp()
    validator = TurkeyHolidayValidator()
  }

  func testTurkeyFixedHolidays() {
    XCTAssertTrue(validator.isHoliday(date: DateComponents(year: 2024, month: 4, day: 23).date!))
  }

  func testTurkeyIslamicHolidays() {
    XCTAssertTrue(validator.isHoliday(date: DateComponents(year: 2024, month: 4, day: 10).date!))
  }
}
